app/policies/g5_updatable/location_policy.rb

```diff
@@ -0,0 +1,34 @@
+module G5Updatable
+  class LocationPolicy < G5Authenticatable::BasePolicy
+    class Scope < G5Authenticatable::BasePolicy::BaseScope
+
+      def resolve
+        return scope.all if has_global_role?
+        scope.where(id: location_roles.map(&:resource_id))
```

Comment:
In addition to considering user's location roles, I think this `resolve` method should also include locations associated with the user's client roles. If a user is assigned a role on Client A, then they should also have access to all of Client A's locations. This business rule holds true across all contexts (that I know of).